### PR TITLE
feat(ingest): refactor mce comparison and add pytest update golden files option

### PR DESCRIPTION
**Pull request by hsheth2**
Also improves the precision of the timestamps for a couple of the tests, and refactors the ignore paths setup.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/tests/integration/feast/test_feast.py [hunk 1]
```diff
@@ -44,8 +44,8 @@ def test_feast_ingest(docker_compose_runner, pytestconfig, tmp_path):
         pipeline.raise_from_status()
 
         # Verify the output.
-        output = mce_helpers.load_json_file(str(tmp_path / "feast_mces.json"))
-        golden = mce_helpers.load_json_file(
-            str(test_resources_dir / "feast_mces_golden.json")
+        mce_helpers.check_golden_file(
+            pytestconfig,
+            output_path=(str(tmp_path / "feast_mces.json")),
```
Comment (kevinhu) [inline metadata-ingestion/tests/integration/feast/test_feast.py:49]: Are the parentheses needed here?

File: metadata-ingestion/tests/integration/bigquery-usage/test_bigquery_usage.py [hunk 2]
```diff
@@ -75,8 +75,8 @@ def test_bq_usage_source(pytestconfig, tmp_path):
         pipeline.run()
         pipeline.raise_from_status()
 
-    output = mce_helpers.load_json_file(str(tmp_path / "bigquery_usages.json"))
-    golden = mce_helpers.load_json_file(
-        str(test_resources_dir / "bigquery_usages_golden.json")
+    mce_helpers.check_golden_file(
+        pytestconfig,
+        output_path=tmp_path / "bigquery_usages.json",
```
Comment (kevinhu) [inline metadata-ingestion/tests/integration/bigquery-usage/test_bigquery_usage.py:80]: Is there a reason why str() was removed here while being retained in other places? Could we move it inside of `check_golden_file`?

File: metadata-ingestion/tests/test_helpers/mce_helpers.py [hunk 3]
```diff
@@ -1,28 +1,50 @@
...
-        # root[0]['proposedSnapshot']['com.linkedin.pegasus2avro.metadata.snapshot.DatasetSnapshot']['aspects'][1]['com.linkedin.pegasus2avro.dataset.UpstreamLineage']['upstreams'][0]['auditStamp']['time']
-        r"root\[\d+\]\['proposedSnapshot'\].+\['aspects'\].+\['created'\]\['time'\]",
-        r"root\[\d+\]\['proposedSnapshot'\].+\['aspects'\].+\['lastModified'\]\['time'\]",
-        r"root\[\d+\]\['proposedSnapshot'\].+\['aspects'\].+\['createStamp'\]\['time'\]",
-        r"root\[\d+\]\['proposedSnapshot'\].+\['aspects'\].+\['auditStamp'\]\['time'\]",
-    }
     diff = deepdiff.DeepDiff(golden, output, exclude_regex_paths=ignore_paths)
-    assert not diff, str(diff)
+    if diff:
+        assert not diff, f"MCEs differ\n{pprint.pformat(diff)}"
```
Comment (kevinhu) [inline metadata-ingestion/tests/test_helpers/mce_helpers.py:35]: Much appreciated!

File: metadata-ingestion/tests/integration/bigquery-usage/test_bigquery_usage.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/tests/integration/bigquery-usage/test_bigquery_usage.py:80]: No strong reason - I'll make it consistent

File: metadata-ingestion/tests/integration/feast/test_feast.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/tests/integration/feast/test_feast.py:49]: nope - will remove

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**